lib/src/button.dart

```diff
@@ -66,6 +73,9 @@ class ImageButton<I> extends StatelessWidget {
     final child = buttonBuilder(context, controller.value.length);
 
     return new GestureDetector(
-        onTap: () => _handlePressed(context), child: child);
+        onTap: () => controller.value.length < this.maxCount
```

Comment:
`maxCount` will either require a default value or a null check should be present here, otherwise Dart will through a null comparison error (e.g. `0 < null`). I would prefer the null check route as assuming default values will always be present or the same can lead to costly refactors.